libs/sdk-ui-dashboard/src/model/commandHandlers/widgets/modifyDrillForInsightWidgetHandler.ts

```diff
@@ -0,0 +1,80 @@
+// (C) 2021 GoodData Corporation
+
+import { DashboardContext } from "../../types/commonTypes";
+import { ModifyDrillsForInsightWidget } from "../../commands";
+import { SagaIterator } from "redux-saga";
+import { put, select } from "redux-saga/effects";
+
+import { DashboardInsightWidgetDrillsModified, insightWidgetDrillsModified } from "../../events/insight";
+import { selectWidgetsMap } from "../../state/layout/layoutSelectors";
+import { validateExistingInsightWidget } from "./validation/widgetValidations";
+import { InsightDrillDefinition } from "@gooddata/sdk-backend-spi";
+import { selectDrillTargetsByWidgetRef } from "../../state/drillTargets/drillTargetsSelectors";
+import { existsDrillDefinitionInArray } from "../../../_staging/drills/InsightDrillDefinitionUtils";
+import { validateInsightDrillDefinition } from "./validation/insightDrillDefinitionValidation";
+import { layoutActions } from "../../state/layout";
+
+export function* modifyDrillForInsightWidgetHandler(
+    ctx: DashboardContext,
+    cmd: ModifyDrillsForInsightWidget,
+): SagaIterator<DashboardInsightWidgetDrillsModified> {
+    const {
+        payload: { drills: drillsToModify },
+        correlationId,
+    } = cmd;
+
+    const widgets: ReturnType<typeof selectWidgetsMap> = yield select(selectWidgetsMap);
+    const insightWidget = validateExistingInsightWidget(widgets, cmd, ctx);
+    const { ref: widgetRef } = insightWidget;
+
+    const selectDrillTargetsByWidgetRefSelector = selectDrillTargetsByWidgetRef(insightWidget.ref);
+
+    const drillTargets: ReturnType<typeof selectDrillTargetsByWidgetRefSelector> = yield select(
+        selectDrillTargetsByWidgetRefSelector,
+    );
+    const { drills: currentInsightDrills } = insightWidget;
+
+    const addedDrillDefinition: InsightDrillDefinition[] = [];
+    const updatedDrillDefinition: InsightDrillDefinition[] = [];
+    const notModifiedDrillDefinition: InsightDrillDefinition[] = [];
+
+    for (const drillItem of drillsToModify) {
+        const validatedDrill = validateInsightDrillDefinition(drillItem, drillTargets, ctx, correlationId);
+
+        if (existsDrillDefinitionInArray(validatedDrill, currentInsightDrills)) {
```

Comment:
This if has not much added value. You do not need to differentiate between added and updated drill definitions so this could be simplified quite a bit

```ts
const addedAndCreatedDrills = drillsToModify.map(drillItem =>
    validateInsightDrillDefinition(drillItem, drillTargets, ctx, correlationId));

const nonModifiedDrills = currentInsightDrills.filter(drillItem =>
    !existsDrillDefinitionInArray(drillItem, drillsToModify));

const updatedInsightDrills = [ ...addedAndCreatedDrills, ...nonModifiedDrills ];

```

Reply:
Refactored